### ISSUE TITLE
pip install fails on OS X/homebrew python3: cchardet error 'define NULL __null'

### ISSUE THREAD

**Issue by rjurney** [datadotworld/data.world-py#109]
# cchardet fails to build on OS X/homebrew python3

Note: versions for OS X/XCode/brew python are latest at the time of posting.

## Error Message after `pip install datadotworld`

```bash
Building wheels for collected packages: cchardet
  Building wheel for cchardet (setup.py) ... error
  ERROR: Complete output from command /Users/rjurney/Software/incubator-superset/homebrew_3.7.3/bin/python3.7 -u -c 'import setuptools, tokenize;__file__='"'"'/private/var/folders/0b/74l_65015_5fcbmbdz1w2xl40000gn/T/pip-install-6v8817va/cchardet/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /private/var/folders/0b/74l_65015_5fcbmbdz1w2xl40000gn/T/pip-wheel-p5c_a5rp --python-tag cp37:
  ERROR: running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.macosx-10.14-x86_64-3.7
  creating build/lib.macosx-10.14-x86_64-3.7/cchardet
  copying src/cchardet/version.py -> build/lib.macosx-10.14-x86_64-3.7/cchardet
  copying src/cchardet/__init__.py -> build/lib.macosx-10.14-x86_64-3.7/cchardet
  running build_ext
  building 'cchardet._cchardet' extension
  creating build/temp.macosx-10.14-x86_64-3.7
  creating build/temp.macosx-10.14-x86_64-3.7/src
  creating build/temp.macosx-10.14-x86_64-3.7/src/cchardet
  creating build/temp.macosx-10.14-x86_64-3.7/src/ext
  creating build/temp.macosx-10.14-x86_64-3.7/src/ext/libcharsetdetect
  creating build/temp.macosx-10.14-x86_64-3.7/src/ext/libcharsetdetect/mozilla
  creating build/temp.macosx-10.14-x86_64-3.7/src/ext/libcharsetdetect/mozilla/extensions
  creating build/temp.macosx-10.14-x86_64-3.7/src/ext/libcharsetdetect/mozilla/extensions/universalchardet
  creating build/temp.macosx-10.14-x86_64-3.7/src/ext/libcharsetdetect/mozilla/extensions/universalchardet/src
  creating build/temp.macosx-10.14-x86_64-3.7/src/ext/libcharsetdetect/mozilla/extensions/universalchardet/src/base
  clang -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/usr/include -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/System/Library/Frameworks/Tk.framework/Versions/8.5/Headers -Isrc/ext/libcharsetdetect/mozilla/extensions/universalchardet/src/base/ -Isrc/ext/libcharsetdetect/nspr-emu/ -Isrc/ext/libcharsetdetect/ -I/usr/local/homebrew/Cellar/python/3.7.3/Frameworks/Python.framework/Versions/3.7/include/python3.7m -c src/cchardet/_cchardet.cpp -o build/temp.macosx-10.14-x86_64-3.7/src/cchardet/_cchardet.o
  src/cchardet/_cchardet.cpp:2351:65: error: too many arguments to function call, expected 3, have 4
      return (*((__Pyx_PyCFunctionFast)meth)) (self, args, nargs, NULL);
             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                     ^~~~
  /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/10.0.1/include/stddef.h:100:18: note: expanded from macro 'NULL'
  #    define NULL __null
                   ^~~~~~
  1 error generated.
  error: command 'clang' failed with exit status 1
  ----------------------------------------
  ERROR: Failed building wheel for cchardet
  Running setup.py clean for cchardet
Failed to build cchardet
Installing collected packages: jsonschema, linear-tsv, ijson, jdcal, et-xmlfile, openpyxl, jsonlines, xlrd, cchardet, tabulator, rfc3986, jsontableschema, pyflakes, pycodestyle, flake8, datapackage, configparser, datadotworld
  Running setup.py install for cchardet ... error
    ERROR: Complete output from command /Users/rjurney/Software/incubator-superset/homebrew_3.7.3/bin/python3.7 -u -c 'import setuptools, tokenize;__file__='"'"'/private/var/folders/0b/74l_65015_5fcbmbdz1w2xl40000gn/T/pip-install-6v8817va/cchardet/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /private/var/folders/0b/74l_65015_5fcbmbdz1w2xl40000gn/T/pip-record-n7ff8pp9/install-record.txt --single-version-externally-managed --compile --install-headers /Users/rjurney/Software/incubator-superset/homebrew_3.7.3/bin/../include/site/python3.7/cchardet:
    ERROR: running install
    running build
    running build_py
    creating build
    creating build/lib.macosx-10.14-x86_64-3.7
    creating build/lib.macosx-10.14-x86_64-3.7/cchardet
    copying src/cchardet/version.py -> build/lib.macosx-10.14-x86_64-3.7/cchardet
    copying src/cchardet/__init__.py -> build/lib.macosx-10.14-x86_64-3.7/cchardet
    running build_ext
    building 'cchardet._cchardet' extension
    creating build/temp.macosx-10.14-x86_64-3.7
    creating build/temp.macosx-10.14-x86_64-3.7/src
    creating build/temp.macosx-10.14-x86_64-3.7/src/cchardet
    creating build/temp.macosx-10.14-x86_64-3.7/src/ext
    creating build/temp.macosx-10.14-x86_64-3.7/src/ext/libcharsetdetect
    creating build/temp.macosx-10.14-x86_64-3.7/src/ext/libcharsetdetect/mozilla
    creating build/temp.macosx-10.14-x86_64-3.7/src/ext/libcharsetdetect/mozilla/extensions
    creating build/temp.macosx-10.14-x86_64-3.7/src/ext/libcharsetdetect/mozilla/extensions/universalchardet
    creating build/temp.macosx-10.14-x86_64-3.7/src/ext/libcharsetdetect/mozilla/extensions/universalchardet/src
    creating build/temp.macosx-10.14-x86_64-3.7/src/ext/libcharsetdetect/mozilla/extensions/universalchardet/src/base
    clang -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/usr/include -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/System/Library/Frameworks/Tk.framework/Versions/8.5/Headers -Isrc/ext/libcharsetdetect/mozilla/extensions/universalchardet/src/base/ -Isrc/ext/libcharsetdetect/nspr-emu/ -Isrc/ext/libcharsetdetect/ -I/usr/local/homebrew/Cellar/python/3.7.3/Frameworks/Python.framework/Versions/3.7/include/python3.7m -c src/cchardet/_cchardet.cpp -o build/temp.macosx-10.14-x86_64-3.7/src/cchardet/_cchardet.o
    src/cchardet/_cchardet.cpp:2351:65: error: too many arguments to function call, expected 3, have 4
        return (*((__Pyx_PyCFunctionFast)meth)) (self, args, nargs, NULL);
               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                     ^~~~
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/10.0.1/include/stddef.h:100:18: note: expanded from macro 'NULL'
    #    define NULL __null
                     ^~~~~~
    1 error generated.
    error: command 'clang' failed with exit status 1
    ----------------------------------------
ERROR: Command "/Users/rjurney/Software/incubator-superset/homebrew_3.7.3/bin/python3.7 -u -c 'import setuptools, tokenize;__file__='"'"'/private/var/folders/0b/74l_65015_5fcbmbdz1w2xl40000gn/T/pip-install-6v8817va/cchardet/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /private/var/folders/0b/74l_65015_5fcbmbdz1w2xl40000gn/T/pip-record-n7ff8pp9/install-record.txt --single-version-externally-managed --compile --install-headers /Users/rjurney/Software/incubator-superset/homebrew_3.7.3/bin/../include/site/python3.7/cchardet" failed with error code 1 in /private/var/folders/0b/74l_65015_5fcbmbdz1w2xl40000gn/T/pip-install-6v8817va/cchardet/
```

## Environment Versions

Mac OS X: Version 10.14.4 Mojave
XCode: Version 10.2.1 (10E1001)
XCode headers and CLI tools are installed

`pip --version`
```
pip 19.1 from /Users/rjurney/Software/incubator-superset/homebrew_3.7.3/lib/python3.7/site-packages/pip (python 3.7)
```

`python --version`
```
Python 3.7.3 # Note: I would like 3.6 but a homebrew recipe for this does not exist
```

`which gcc`
```bash
/usr/bin/gcc
```

`rjurney$ gcc --version`
```bash
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
Apple LLVM version 10.0.1 (clang-1001.0.46.4)
Target: x86_64-apple-darwin18.5.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

:(

**Comment by rjurney** [comment]
At this point I installed gcc 8.3 via `brew upgrade gcc` and then `ln -s /usr/local/homebrew/bin/gcc-8 /usr/local/homebrew/bin/gcc` to get it in my path as the preferred gcc.

It had no effect, the same error appears.

**Comment by laconc** [comment]
@rjurney this has been resolved in v1.7.0, with upgraded dependencies.
